monitoring/dashboard.py

```diff
@@ -525,6 +559,167 @@ def top10_errors_by_bucket(title, code):
     title="5xx : Top10 by Bucket", code='~"5.."'
 )
 
+
+quotaHealth = TimeSeries(
+    title="Quota service uptime",
+    dataSource="${DS_PROMETHEUS}",
+    lineInterpolation="stepAfter",
+    fillOpacity=30,
+    unit=UNITS.PERCENT_FORMAT,
+    targets=[Target(
+        expr="\n".join([
+            'sum(rate(s3_cloudserver_utilization_service_state{namespace="${namespace}", state="healthy"}[$__rate_interval])) * 100',  # noqa: E501
+            "   /",
+            'sum(rate(s3_cloudserver_utilization_service_state{namespace="${namespace}", state=~"unhealthy|healthy"}[$__rate_interval]))',  # noqa: E501
+        ]),
+        legendFormat="State",
+    )],
+    thresholds=[
+        Threshold("green", 0, 95.0),
+        Threshold("orange", 1, 90.0),
+        Threshold("red", 2, 0.0),
+    ],
+)
+
+
+quotaStatusCode = TimeSeries(
+    title="Quota evaluation status code over time",
+    dataSource="${DS_PROMETHEUS}",
+    fillOpacity=30,
+    lineInterpolation="smooth",
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by (code) (rate(s3_cloudserver_quota_evaluation_duration_seconds_count{namespace="${namespace}", code=~"2.."}[$__rate_interval]))',  # noqa: E501
+        legendFormat="Success",
+    ), Target(
+        expr='sum by (code) (rate(s3_cloudserver_quota_evaluation_duration_seconds_count{namespace="${namespace}", code="429"}[$__rate_interval]))',  # noqa: E501
+        legendFormat="Quota Exceeded",
+    ), Target(
+        expr='sum by (code) (rate(s3_cloudserver_quota_evaluation_duration_seconds_count{namespace="${namespace}", code=~"4..", code!="429"}[$__rate_interval]))',  # noqa: E501
+        legendFormat="Status 4xx",
+    ), Target(
+        expr='sum by (code) (rate(s3_cloudserver_quota_evaluation_duration_seconds_count{namespace="${namespace}", code=~"5.."}[$__rate_interval]))',  # noqa: E501
+        legendFormat="Status 5xx",
```

Comment:
are these 4xx and 5xx status relevant for this metric? at the end of the evaluation, we can only have either 'proceed' or '429'... Even if the query eventually ends up with another error code, the metric will have already recorded either 200 or 429...

Reply:
Removed.